### DISCUSSION TITLE
通过手工配置fprint+fwupd支持更多指纹设备

### DISCUSSION THREAD

**Discussion by ziggy1030** [linuxdeepin/developer-center#3409] [Show and tell | 成果展示]
起因：检查fprint的wiki有支持我的指纹设备id，更新libfprint等相关组件为最新版本后，仍然无法识别指纹设备
来源：搜索device id后发现有发行版支持，提及了fwupd固件问题，经试验后本人hp probook 455 G7的Synaptics指纹设备成功激活，固有下文解决方案
#Debian上游最新fwupd版本较老，本文采用flatpak较新版本替代

#只能提升能够使用设备指纹功能的概率，并不保证成功率，有更好的方案欢迎共享

#目前只测试UOS，其他发行版理论上通用

#需要有一定命令操作能力和耐心要求，喜欢鼠标点点点的可以考虑绕道了

#基础原理:frpint相关组件提供支持，fwupd更新硬件固件

# 前提准备

1.fprint相关组件包最新版本（附件）

2.提前安装flatpak:以root权限更新源后执行apt install flatpak

3.确保指纹设备无故障，其他发行版/OS上正常运行

4.执行lsusb获取指纹设备的device id，到https://fprint.freedesktop.org/supported-devices.html上搜索是否在support list内,**不在列表大概率不支持，可以忽略下面内容了**
![截图_选择区域_20221025162236](https://user-images.githubusercontent.com/39146681/197769679-46f899ca-9888-4887-9797-a6ad4f001087.png)

# 基础组件安装

1.使用apt同时安装附件内所有包

安装方法:root权限下执行apt install xxx本地包，即可以apt形式安装本地debian包

2.(可选)为flatpak换为国内源:

方法一:flatpak remote-add --if-not-exists sjtu  https://mirror.sjtu.edu.cn/flathub/flathub.flatpakrepo

方法二:

flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo

flatpak remote-modify flathub --url=https://mirror.sjtu.edu.cn/flathub

参考文档:https://skiy.net/d/211

3.非root账户下运行flatpak install flathub org.freedesktop.fwupd

# 安装fprint后测试

1.执行fprintd-enroll,正常调用识别则可以重启直接使用，显示无效设备则需要进行下一板块

# 更新固件

1.root权限下执行flatpak run org.freedesktop.fwupd get-devices扫描设备
![fwupd](https://user-images.githubusercontent.com/39146681/197769816-6415981e-c564-4e89-bf75-517f5dcd7a5f.png)

2.记录指纹设备的device id或guid
![fwupd_marked](https://user-images.githubusercontent.com/39146681/197769841-59eeb84a-ad9f-4055-aac9-ac11eecacd54.png)

3.root权限下执行flatpak run org.freedesktop.fwupd refresh --force刷新可更新固件

4.root权限下执行flatpak run org.freedesktop.fwupd update [device ID/GUID]仅更新指纹设备的固件

# 验证

1.执行fprintd-enroll,正常调用识别则可以重启直接使用

2.不通过重新安装附件内所有组件包，并重新更新固件

# 避雷

1.fprintd-enroll添加完后必须使用fprintd-list [username]和fprintd-delete [username]删除创建的指纹数据，否则控制中心生物验证无法使用

2.重启后控制中心生物验证方可使用

3.安装完fprint包后原则上不要随意卸载依赖，尽管被标记为“不需要”，也不要执行自动清理依赖。若突然无法使用，重新执行上述步骤即可。
完整文档:
[fprint_cn.md](https://github.com/linuxdeepin/developer-center/files/9860101/fprint_cn.md)
[fprint_en.md](https://github.com/linuxdeepin/developer-center/files/9860102/fprint_en.md)

ges.zip](https://github.com/linuxdeepin/developer-center/files/9490226/frpint_packages.zip)

**Comment by ziggy1030** [comment]
感觉这个问题还扯出来了dde有没有需求在驱动管理那加上fwupd的功能，其他发行版貌似有集成到系统功能内

**Comment by aleemont1** [comment]
Waiting for EN version :)

**Reply by ziggy1030** [reply]
en version here:https://github.com/linuxdeepin/developer-center/discussions/3476en version here:https://github.com/linuxdeepin/developer-center/discussions/3476

**Comment by kondors1995** [comment]
Looks intresting
